code/modules/hydroponics/grown/banana.dm

```diff
@@ -115,6 +116,8 @@
 /obj/item/weapon/grown/bananapeel/bluespace/Crossed(AM)
 	if(..())
 		var/teleport_radius = potency / 10
+		if(teleport_radius < 1) //not enough potency to teleport
+			return
 		do_teleport(AM, get_turf(AM), teleport_radius)
```

Comment:
technically, it doesn't fix the infinite loop, it only makes it incredibly unlikely. You can still be teleported on the exact same tile and have 50% of being teleported again if the peel isn't deleted. Repeat endlessly.